crates/bevy_reflect/src/utility.rs

```diff
@@ -239,22 +239,38 @@ impl<T: TypedProperty> GenericTypeCell<T> {
         G: Any + ?Sized,
         F: FnOnce() -> T::Stored,
```

Comment:
it might require some more elaborate changes, but i'm actually pretty sure we can elide `F` here too. i'm pretty sure the functions used in the derive macro are pure, so this could just take an `f: fn() -> T::Stored` parameter

Reply:
`FnXxx` traits don't act any differently to `fn` types in considering generics. everything should be fine.

Reply:
> `FnXxx` traits don't act any differently to `fn` types in considering generics. everything should be fine.

Good to know! Then yeah I'd be curious to see how much further this would get us.

Maybe this could be explored as a followup PR? I don't think we need to block the current improvements getting merged in.

Reply:
https://users.rust-lang.org/t/question-about-accepting-closures/29405

Reply:
Pretty sure we need to use `fn()` pointers to avoid monomorphization

Reply:
> > FnXxx traits don't act any differently to fn types in considering generics. everything should be fine.
> 
> 
> 
> Hmm I don't understand how that could be true, as an `impl FnXxx` doesn't have a fixed size because it could be a closure containing any type/amount of data?

I think this was in reference to my question about the function referencing generic types and `Self` from its environment

Reply:
yeah i wasn't very clear i was just stating that `fn`s and `Fn`s can both be monomorphized.

Reply:
Ahhhh no you were clear I was just being dense :)

Reply:
Yeah I believe that's a further optimisation that we could add, but it would require changing the public facing API for this type, so I'd rather save it for a follow-up PR. I think the scope of this PR being a slight refactor for improved binary sizes is worth preserving.